bridge_adaptivity/module/models.py

```diff
@@ -70,23 +87,48 @@ def save(self, *args, **kwargs):
         super(SequenceItem, self).save(*args, **kwargs)
 
 
+@python_2_unicode_compatible
+class GradingPolicy(ModelWithDefaultInstanceMixin, models.Model):
+    """Predefined set of Grading policy objects. Define how to grade collections."""
+
+    name = models.CharField(max_length=255)
+    public_name = models.CharField(max_length=255)
+    threshold = models.PositiveIntegerField(blank=True, default=0, help_text="Grade policy: 'Q'")
+    is_default = models.BooleanField(default=False)
+
+    def _points_earned_grade(self, trials_count, points_earned):
+        return points_earned / max(self.threshold, trials_count)
+
+    def _trials_count(self, trials_count):
+        return trials_count / max(self.threshold, trials_count)
+
+    def calculate_grade(self, sequence):
+        items_result = sequence.items.aggregate(points_earned=Sum('score'), trials_count=Count('score'))
+        grading_map = {
+            # name, function
+            'points_earned': (self._points_earned_grade, (items_result['trials_count'], items_result['points_earned'])),
+            'trials_count': (self._trials_count, (items_result['trials_count'],))
+        }
+        if self.name in grading_map:
```

Comment:
L112-113 won't be needed if name will be checked from the proposed choices see my comment to the L94